scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/shared/PortalConfig.java

```diff
@@ -0,0 +1,207 @@
+/*
+ * ProActive Parallel Suite(TM):
+ * The Open Source library for parallel and distributed
+ * Workflows & Scheduling, Orchestration, Cloud Automation
+ * and Big Data Analysis on Enterprise Grids & Clouds.
+ *
+ * Copyright (c) 2007 - 2017 ActiveEon
+ * Contact: contact@activeeon.com
+ *
+ * This library is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation: version 3 of
+ * the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.scheduler.shared;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import org.ow2.proactive_grid_cloud_portal.common.client.json.JSONException;
+import org.ow2.proactive_grid_cloud_portal.common.client.json.JSONUtils;
+
+import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONBoolean;
+import com.google.gwt.json.client.JSONObject;
+import com.google.gwt.json.client.JSONString;
+
+
+/**
+ * @author ActiveEon Team
+ * @since Feb 27, 2017
+ */
+public class PortalConfig {
+
+    /**
+     * Logger
+     */
+    private static final Logger LOGGER = Logger.getLogger(PortalConfig.class.getName());
+
+    /**
+     * Path to the user-defined portal property file
+     */
+    public static final String CONFIG_PATH = "scheduler-portal.conf";
+
+    /**
+     * Singleton behaviour
+     */
+    private static PortalConfig instance = null;
+
+    /**
+     * Config properties
+     */
+    private Map<String, String> properties = new HashMap<>();
+
+    /**
+     * Config properties
+     */
+    private List<JSONColumn> extraColumns = null;
+
+    /**
+     * @return current static config instance, cannot be null
+     */
+    public static PortalConfig get() {
+        if (instance == null) {
+            instance = new PortalConfig();
+        }
+        return instance;
+    }
+
+    public List<JSONColumn> getExtraColumns() {
+
+        if (extraColumns == null) {
+            String extraColumnsJson = properties.get("execution-list-extra-columns");
```

Comment:
Can "execution-list-extra-columns" be a private final static string in this class?